pkg/agent/endpoints/peertracker.go

```diff
@@ -19,6 +19,7 @@ func (a PeerTrackerAttestor) Attest(ctx context.Context) ([]*common.Selector, er
 	if !ok {
 		return nil, status.Error(codes.Internal, "peer tracker watcher missing from context")
 	}
+	defer watcher.Close()
```

Comment:
I'm not sure this is how we want to tackle this. This change would cause the watcher to be closed unconditionally after the first attestation. This would preclude us from redoing workload attestation, which we've discussed in the past...

The watcher lifetime is currently managed by the gRPC connection. If we want to divorce that, I wonder if we should do it internally on a failed "IsAlive" check, or externally when IsAlive fails below....

Reply:
> The watcher lifetime is currently managed by the gRPC connection. If we want to divorce that, I wonder if we should do it internally on a failed "IsAlive" check, or externally when IsAlive fails below....

Initially, I put the close explcitly only in the `IsAlive` error block below. I don't think my logic for moving it to this defer was sound. Also, the leak is probably more due to the grpc conns than the open files in `/proc`. So far this would only address the `procfd`.

Reply:
I am having trouble figuring out where/how to get at the underlying connection during the peertracker `IsAlive` to close it b/c I think that is actually what we want to do when `IsAlive` fails and that should also close the `Watcher`. I see the wrapped `Accept` in `common/peertracker/listener.go`, but not how to get the resulting conn where we should close it.

Reply:
@azdagron Should we try to merge this as-is or should we continue pursuing a force-close of conns on failure of `IsAlive`? I wasn't clear after discussion in contributor sync. If we should continue to pursue the force-close, I will need some pointers/help. I did not see a way to plumb the conns to close them in `IsAlive` or at the call site.